rtl/eclrtl/rtlformat.cpp

```diff
@@ -1402,17 +1404,22 @@ unsigned CommonCSVWriter::getChildrenMaxColumnID(CCSVItem* item, unsigned& maxCo
     return maxColumnID;
 }
 
-void CommonCSVWriter::escapeQuoted(unsigned len, char const* in, StringBuffer& out)
+void CommonCSVWriter::escapeString(unsigned len, char const* in, StringBuffer& out)
 {
     char const* finger = in;
     while (len--)
     {
-        //RFC-4180, paragraph "If double-quotes are used to enclose fields, then a double-quote
-        //appearing inside a field must be escaped by preceding it with another double quote."
-        //unsigned newLen = 0;
-        if (*finger == '"')
-            out.append('"');
-        out.append(*finger);
+        if (*finger == '\0')
```

Comment:
I would reverse this test - simpler to have the common (non null) branch 1st (and the compiler will also assume that is the case).
This should also have a comment e.g.
//null bytes should not occur, but encode them if they do to prevent problems elsewhere